src/main/java/com/conveyal/gtfs/validator/FlexValidator.java

```diff
@@ -129,188 +164,273 @@ public static List<NewGTFSError> validateStopArea(
             );
         }
         return errors;
     }
 
     /**
-     * Check that a stop time conforms to flex specification constraints.
+     * Check that a stop time fields conform to flex specification constraints.
      */
-    public static List<NewGTFSError> validateStopTime(
-        StopTime stopTime,
-        List<StopArea> stopAreas,
-        List<Location> locations
-    ) {
-
+    public static List<NewGTFSError> validateStopTime(StopTime stopTime) {
         List<NewGTFSError> errors = new ArrayList<>();
+        validateArrivalTime(stopTime, errors);
+        validateDepartureTime(stopTime, errors);
+        validateStopId(stopTime, errors);
+        validateLocationGroupId(stopTime, errors);
+        validateLocationId(stopTime, errors);
+        validateStartPickupDropOffWindow(stopTime, errors);
+        validateEndPickupDropOffWindow(stopTime, errors);
+        validatePickUpType(stopTime, errors);
+        validateDropOffType(stopTime, errors);
+        validateContinuousPickup(stopTime, errors);
+        validateContinuousDropOff(stopTime, errors);
+        return errors;
+    }
 
-        if (stopTime.arrival_time != INT_MISSING &&
-            (stopTime.start_pickup_drop_off_window != INT_MISSING ||
-                stopTime.end_pickup_drop_off_window != INT_MISSING)
+    /**
+     * Conditionally Required:
+     * - Forbidden when start_pickup_drop_off_window or end_pickup_drop_off_window are defined.
+     * - Optional otherwise.
+     */
+    public static void validateArrivalTime(StopTime stopTime, List<NewGTFSError> errors) {
+        if (
+            stopTime.arrival_time != INT_MISSING &&
+            (stopTime.start_pickup_drop_off_window != INT_MISSING || stopTime.end_pickup_drop_off_window != INT_MISSING)
         ) {
-            // Arrival time must not be defined if start/end pickup drop off window is defined.
-            errors.add(NewGTFSError.forEntity(
-                stopTime,
-                NewGTFSErrorType.FLEX_FORBIDDEN_ARRIVAL_TIME).setBadValue(Integer.toString(stopTime.arrival_time))
+            errors.add(NewGTFSError
+                .forEntity(stopTime, NewGTFSErrorType.FLEX_FORBIDDEN_ARRIVAL_TIME)
+                .setBadValue(Integer.toString(stopTime.arrival_time))
             );
         }
-        if (stopTime.departure_time != INT_MISSING &&
-            (stopTime.start_pickup_drop_off_window != INT_MISSING ||
-                stopTime.end_pickup_drop_off_window != INT_MISSING)
+    }
+
+    /**
+     * Conditionally Required:
+     * - Forbidden when start_pickup_drop_off_window or end_pickup_drop_off_window are defined.
+     * - Optional otherwise.
+     */
+    public static void validateDepartureTime(StopTime stopTime, List<NewGTFSError> errors) {
+        if (
+            stopTime.departure_time != INT_MISSING &&
+            (stopTime.start_pickup_drop_off_window != INT_MISSING || stopTime.end_pickup_drop_off_window != INT_MISSING)
```

Comment:
Extract a method for checking that at least one of the start/end pickup window fields is missing (a search for that subcondition returns ~7 results).